patches/server/1049-Fix-CraftBukkit-drag-system.patch

```diff
@@ -0,0 +1,263 @@
++                                this.setCarried(slot.safeInsert(this, itemstack3, i2, isSingleSlotDrag)); // Paper - Fix CraftBukkit drag system
+                             }
+                         } else if (slot.mayPickup(player)) {
+                             if (itemstack3.isEmpty()) {
+@@ -567,7 +567,7 @@ public abstract class AbstractContainerMenu {
+                             } else if (slot.mayPlace(itemstack3)) {
+                                 if (ItemStack.isSameItemSameComponents(itemstack, itemstack3)) {
+                                     i2 = clickaction == ClickAction.PRIMARY ? itemstack3.getCount() : 1;
+-                                    this.setCarried(slot.safeInsert(itemstack3, i2));
++                                    this.setCarried(slot.safeInsert(this, itemstack3, i2, isSingleSlotDrag)); // Paper - Fix CraftBukkit drag system
+                                 } else if (itemstack3.getCount() <= slot.getMaxStackSize(itemstack3)) {
+                                     this.setCarried(itemstack);
+                                     slot.setByPlayer(itemstack3);
+@@ -680,10 +680,10 @@ public abstract class AbstractContainerMenu {
+ 
+     }
+ 
+-    private boolean tryItemClickBehaviourOverride(Player player, ClickAction clickType, Slot slot, ItemStack stack, ItemStack cursorStack) {
++    private boolean tryItemClickBehaviourOverride(Player player, ClickAction clickType, Slot slot, ItemStack stack, ItemStack cursorStack, boolean isSingleSlotDrag) { // Paper - Fix CraftBukkit drag system
+         FeatureFlagSet featureflagset = player.level().enabledFeatures();
+ 
+-        return cursorStack.isItemEnabled(featureflagset) && cursorStack.overrideStackedOnOther(slot, clickType, player) ? true : stack.isItemEnabled(featureflagset) && stack.overrideOtherStackedOnMe(cursorStack, slot, clickType, player, this.createCarriedSlotAccess());
++        return cursorStack.isItemEnabled(featureflagset) && cursorStack.overrideStackedOnOther(slot, clickType, player, isSingleSlotDrag) ? true : stack.isItemEnabled(featureflagset) && stack.overrideOtherStackedOnMe(cursorStack, slot, clickType, player, this.createCarriedSlotAccess()); // Paper - Fix CraftBukkit drag system
+     }
+ 
+     private SlotAccess createCarriedSlotAccess() {
+diff --git a/src/main/java/net/minecraft/world/inventory/Slot.java b/src/main/java/net/minecraft/world/inventory/Slot.java
+index c39c773112fb8b534b926f2f2b47fe6fbb69fcb2..6c8a4d8ba0d27e6e5e58ec1f39b03ac6cd1f17ae 100644
+--- a/src/main/java/net/minecraft/world/inventory/Slot.java
++++ b/src/main/java/net/minecraft/world/inventory/Slot.java
+@@ -7,6 +7,7 @@ import net.minecraft.resources.ResourceLocation;
+ import net.minecraft.world.Container;
+ import net.minecraft.world.entity.player.Player;
+ import net.minecraft.world.item.ItemStack;
++import org.bukkit.craftbukkit.inventory.CraftItemStack; // Paper - Fix CraftBukkit drag system
```

Comment:
why is this import necessary?

Reply:
cause i would have to do a fqi like 5 times in each class